### PR TITLE
chore: use positional parameters for cleanliness

### PR DESCRIPTION
**Pull request by Mercy811**
Some cleanups 
- Use positional parameters for cleanliness. Thanks @justin-fiedler for calling it out https://github.com/amplitude/amplitude-dev-center/pull/239#discussion_r1525165051
  - Positional parameters are required by default. Use `[]` to make it optional
  - `{}` indicates named parameters, which are optional by default. Use `required` to make it required. 
```
// Before
track({ required BaseEvent event, EventOptions? options, })
// Usage
track(event: BaseEvent(eventType:'BUTTON_CLICKED', sessionId: -1))

// After
track( BaseEvent event, [ EventOptions? options, ])
// Usage
track(BaseEvent(eventType:'BUTTON_CLICKED', sessionId: -1))
```
- Add more rules to flutter static analyzer in `analysis_options.yaml`
  - Use recommend rules https://github.com/dart-lang/lints/blob/main/lib/recommended.yaml. `flutter analyze` use core rules by default
  - Add [prefer_single_quotes](https://dart.dev/tools/linter-rules/prefer_single_quotes#:~:text=DO%20use%20single%20quotes%20where,escaped%20apostrophe%20is%20not%20flagged) 
  - Add [prefer_relative_imports](https://dart.dev/tools/linter-rules/prefer_relative_imports)

### CODE REVIEW

**Review by justin-fiedler — APPROVED** [review]
LGTM. Thanks @Mercy811

File: lib/identify.dart [hunk 1]
```diff
@@ -3,15 +3,25 @@ import 'package:flutter/foundation.dart';
...
   static const String OP_UNSET = r'$unset';
+  // ignore: constant_identifier_names
   static const String OP_PREPEND = r'$prepend';
+  // ignore: constant_identifier_names
   static const String OP_PREINSERT = r'$preInsert';
+  // ignore: constant_identifier_names
   static const String OP_POSTINSERT = r'$postInsert';
+  // ignore: constant_identifier_names
   static const String OP_REMOVE = r'$remove';
+  // ignore: constant_identifier_names
```
Comment (yuhao900914) [inline lib/identify.dart:24]: Curious about what's the `ignore` comment for?

**Review by yuhao900914 — APPROVED** [review]
LGTM.

**Review by qingzhuozhen — APPROVED** [review]

File: lib/identify.dart (hunk 1 above)
Comment (Mercy811) [reply lib/identify.dart:24]: It's to ignore constant_identifier_names lint rule.

**Pull request merged**